pkgs/development/haskell-modules/configuration-ghc-9.0.x.nix

```diff
@@ -104,14 +104,69 @@ self: super: {
   # https://github.com/Soostone/retry/issues/71
   retry = dontCheck super.retry;
 
-  # hlint 3.3 needs a ghc-lib-parser newer than the one from stackage
-  hlint = super.hlint_3_3_4.overrideScope (self: super: {
-    ghc-lib-parser = overrideCabal self.ghc-lib-parser_9_0_1_20210324 {
-      doHaddock = false;
-    };
-    ghc-lib-parser-ex = self.ghc-lib-parser-ex_9_0_0_4;
-  });
+  hlint = super.hlint_3_3_4;
 
   # pick right version for compiler
   ghc-api-compat = doDistribute super.ghc-api-compat_9_0_1;
+  ghc-lib-parser = overrideCabal self.ghc-lib-parser_9_0_1_20210324 {
+    doHaddock = false;
+  };
+  ghc-lib-parser-ex = self.ghc-lib-parser-ex_9_0_0_4;
+  ghc-lib = self.ghc-lib_9_0_1_20210324;
+
+  # Unbumped base bound
+  semialign = doJailbreak super.semialign;
+
+  # Will probably be needed for brittany support
+  # https://github.com/lspitzner/czipwith/pull/2
+  #czipwith = appendPatch super.czipwith
+  #    (pkgs.fetchpatch {
+  #      url = "https://github.com/lspitzner/czipwith/commit/b6245884ae83e00dd2b5261762549b37390179f8.patch";
+  #      sha256 = "08rpppdldsdwzb09fmn0j55l23pwyls2dyzziw3yjc1cm0j5vic5";
+  #    });
+
+  # https://github.com/mokus0/th-extras/pull/8
+  th-extras = overrideCabal super.th-extras (old: {
+      src = pkgs.fetchFromGitHub  {
+        owner = "mokus0";
+        repo = "th-extras";
+        rev = "0d050b24ec5ef37c825b6f28ebd46787191e2a2d";
+        sha256 = "045f36yagrigrggvyb96zqmw8y42qjsllhhx2h20q25sk5h44xsd";
+      };
+      libraryHaskellDepends = old.libraryHaskellDepends ++ [self.th-abstraction];
+    });
+
+  dependent-sum-template = dontCheck (appendPatch super.dependent-sum-template
+      (pkgs.fetchpatch {
+        url = "https://github.com/obsidiansystems/dependent-sum/commit/8cf4c7fbc3bfa2be475a17bb7c94a1e1e9a830b5.patch";
+        sha256 = "02wyy0ciicq2x8lw4xxz3x5i4a550mxfidhm2ihh60ni6am498ff";
+        stripLen = 2;
+        extraPrefix = "";
+      }));
+
+  hiedb = overrideCabal super.hiedb (old: {
+    libraryHaskellDepends = old.libraryHaskellDepends ++ [self.ghc-api-compat];
```

Comment:
Is this hidden behind a flag or why doesn't cabal2nix pick up on this?

Reply:
Yes: https://github.com/wz1000/HieDb/blob/0e660aea0e2cf7a64c125b396b0229fcf679e6f7/hiedb.cabal#L74